File: src/main/java/net/gtaun/shoebill/common/command/CommandEntry.java
package net.gtaun.shoebill.common.command;

import net.gtaun.shoebill.common.command.CommandEntryInternal.CommandHandlerInternal;
import net.gtaun.shoebill.object.Player;

public class CommandEntry
{
	private CommandEntryInternal entry;
	private String path;
	
	
	public CommandEntry(CommandEntryInternal entry, String path)
	{
		this.entry = entry;
		this.path = path;
	}
	
	public String getPath()
	{
		return path;
	}
	
	public String getCommand()
	{
		return entry.completeCommand(path);
	}
	
	public Class<?>[] getParamTypes()
	{
		return entry.getParamTypes();
	}

	public String[] getParamNames()
	{
		return entry.getParamNames();
	}

	public short getPriority()
	{
		return entry.getPriority();
	}

	public String getHelpMessage() { return entry.getHelpMessage(); }

	public boolean isCaseSensitive() { return entry.isCaseSensitive(); }

	public CommandHandlerInternal getHandler()
	{
		return entry.getHandler();
	}
	
	public boolean handle(Player player, Object[] params)
	{
		return entry.handle(player, params);
	}

	public String getCategorie() { return entry.getCategorie(); }
}

File: src/main/java/net/gtaun/shoebill/common/command/CommandEntry.kt
package net.gtaun.shoebill.common.command

import net.gtaun.shoebill.common.command.CommandEntryInternal.CommandHandlerInternal
import net.gtaun.shoebill.entities.Player

class CommandEntry(val entry: CommandEntryInternal, val path: String) {

    val command: String
        get() = entry.completeCommand(path)

    val parameterTypes: Array<Class<*>>
        get() = entry.paramTypes

    val priority: Short
        get() = entry.priority

    val helpMessage: String?
        get() = entry.helpMessage

    val isCaseSensitive: Boolean
        get() = entry.isCaseSensitive

    val handler: CommandHandlerInternal
        get() = entry.handler

    fun handle(player: Player, params: Array<Any>): Boolean = entry.handle(player, params)

    val category: String?
        get() = entry.category

    val parameters: Array<CommandParameter>
        get() = entry.parameterAnnotations
}
